vllm/v1/metrics/loggers.py

```diff
@@ -116,9 +131,42 @@ def log(self, scheduler_stats: SchedulerStats,
         self.counter_generation_tokens.inc(
             iteration_stats.num_generation_tokens)
 
+        for finished_request in iteration_stats.finished_requests:
+            self.histogram_num_prompt_tokens_request.observe(
```

Comment:
QQ, in V0, we did:

```bash
histogram.labels(**self.labels).observe(datum)
```

https://github.com/vllm-project/vllm/blob/0f657bdc52d4ad1d079beddf8e7556c419aca7b4/vllm/engine/metrics.py#L541

Do you know why this is or is not needed?

Reply:
Ill hit automerge to unblock you

Reply:
`labels()` is a [pretty expensive factory method](https://github.com/prometheus/client_python/blob/46eae7bae88f76951f7246d9f359f2dd5eeff110/prometheus_client/metrics.py#L134) and our labels aren't ever changing, so I'm building these labelled metrics in the `PrometheusStatLogger` constructor

```
self.histogram_num_generation_tokens_request = \
            prometheus_client.Histogram(
                name="vllm:request_generation_tokens",
                documentation="Number of generation tokens processed.",
                buckets=build_1_2_5_buckets(max_model_len),
                labelnames=labelnames).labels(*labelvalues)
```

Not so easy to spot it though!